tests/type_propagation/block-local-test.toit

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2024 Toitware ApS.
+// Use of this source code is governed by a Zero-Clause BSD license that can
+// be found in the tests/LICENSE file.
+
+main:
+  test-single
+  test-multi
+  test-outer
+  test-recursive
```

Comment:
Add test where a function gets two blocks and they influence each other's types.

Reply:
Done.